net/ipfamily_test.go

```diff
@@ -0,0 +1,418 @@
+		family := IPStringFamily(testCases[i].ip)
+		isIPv4 := IsIPv4String(testCases[i].ip)
+		isIPv6 := IsIPv6String(testCases[i].ip)
+		switch testCases[i].family {
+		case IPFamilyUnknown:
+			if family != IPFamilyUnknown || isIPv4 || isIPv6 {
+				t.Errorf("[%d] Expected family %q, isIPv4 %v, isIPv6 %v. Got family %q, isIPv4 %v, isIPv6 %v", i+1, IPFamilyUnknown, false, false, family, isIPv4, isIPv6)
+			}
+		case IPv4:
+			if family != IPv4 || !isIPv4 || isIPv6 {
+				t.Errorf("[%d] Expected family %q, isIPv4 %v, isIPv6 %v. Got family %q, isIPv4 %v, isIPv6 %v", i+1, IPv4, true, false, family, isIPv4, isIPv6)
+			}
+		case IPv6:
+			if family != IPv6 || isIPv4 || !isIPv6 {
+				t.Errorf("[%d] Expected family %q, isIPv4 %v, isIPv6 %v. Got family %q, isIPv4 %v, isIPv6 %v", i+1, IPv6, false, true, family, isIPv4, isIPv6)
+			}
+		}
+	}
+}
+
+func TestIPFamily(t *testing.T) {
+	testCases := []struct {
+		ip     net.IP
+		family IPFamilyType
+	}{
+		{
+			ip:     net.IPv4zero,
+			family: IPv4,
+		},
+		{
+			ip:     net.IPv4bcast,
+			family: IPv4,
+		},
+		{
+			ip:     ParseIPSloppy("127.0.0.1"),
+			family: IPv4,
+		},
+		{
+			ip:     ParseIPSloppy("10.20.40.40"),
+			family: IPv4,
+		},
+		{
+			ip:     ParseIPSloppy("172.17.3.0"),
+			family: IPv4,
+		},
+		{
+			ip:     nil,
+			family: IPFamilyUnknown,
+		},
+		{
+			ip:     net.IPv6loopback,
+			family: IPv6,
+		},
+		{
+			ip:     net.IPv6zero,
+			family: IPv6,
+		},
+		{
+			ip:     ParseIPSloppy("fd00::600d:f00d"),
+			family: IPv6,
+		},
+		{
+			ip:     ParseIPSloppy("2001:db8::5"),
+			family: IPv6,
+		},
+	}
+	for i := range testCases {
+		family := IPFamily(testCases[i].ip)
+		isIPv4 := IsIPv4(testCases[i].ip)
+		isIPv6 := IsIPv6(testCases[i].ip)
+		switch testCases[i].family {
+		case IPFamilyUnknown:
+			if family != IPFamilyUnknown || isIPv4 || isIPv6 {
+				t.Errorf("[%d] Expected family %q, isIPv4 %v, isIPv6 %v. Got family %q, isIPv4 %v, isIPv6 %v", i+1, IPFamilyUnknown, false, false, family, isIPv4, isIPv6)
+			}
+		case IPv4:
+			if family != IPv4 || !isIPv4 || isIPv6 {
+				t.Errorf("[%d] Expected family %q, isIPv4 %v, isIPv6 %v. Got family %q, isIPv4 %v, isIPv6 %v", i+1, IPv4, true, false, family, isIPv4, isIPv6)
+			}
+		case IPv6:
+			if family != IPv6 || isIPv4 || !isIPv6 {
+				t.Errorf("[%d] Expected family %q, isIPv4 %v, isIPv6 %v. Got family %q, isIPv4 %v, isIPv6 %v", i+1, IPv6, false, true, family, isIPv4, isIPv6)
+			}
+		}
+	}
+}
+
+func TestCIDRFamily(t *testing.T) {
+	testCases := []struct {
+		desc   string
+		cidr   string
+		family IPFamilyType
+	}{
+		{
+			desc:   "ipv4 CIDR 1",
+			cidr:   "10.0.0.0/8",
+			family: IPv4,
+		},
+		{
+			desc:   "ipv4 CIDR 2",
+			cidr:   "192.168.0.0/16",
+			family: IPv4,
+		},
+		{
+			desc:   "ipv6 CIDR 1",
+			cidr:   "::/1",
+			family: IPv6,
+		},
+		{
+			desc:   "ipv6 CIDR 2",
+			cidr:   "2000::/10",
+			family: IPv6,
+		},
+		{
+			desc:   "ipv6 CIDR 3",
+			cidr:   "2001:db8::/32",
+			family: IPv6,
+		},
```

Comment:
test case for unknown?
```suggestion
		},{
			desc:   "Unknown IP family",
			cidr:   "invalid.cidr/mask",
			family: IPFamilyUnknown,
		},
		
```

Reply:
sure. (the tests that are in the PR now are just a consolidation of the pre-existing tests, which did not have any unknown/nil tests for CIDR)